diff --git a/CMakeLists.txt b/CMakeLists.txt
index b8aad0d783..c7f3d313f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1666,9 +1666,10 @@ target_link_libraries(gpr
   absl::variant
 )
 if(_gRPC_PLATFORM_ANDROID)
+  find_library(log-lib log )
   target_link_libraries(gpr
     android
-    log
+    ${log-lib}
   )
 endif()
diff --git a/cmake/libprotobuf-lite.cmake b/cmake/libprotobuf-lite.cmake
index 83e970312..4fcf666be 100644
--- a/cmake/libprotobuf-lite.cmake
+++ b/cmake/libprotobuf-lite.cmake
@@ -102,7 +102,8 @@ if(protobuf_LINK_LIBATOMIC)
   target_link_libraries(libprotobuf-lite PRIVATE atomic)
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
-  target_link_libraries(libprotobuf-lite PRIVATE log)
+  find_library(log-lib log )
+  target_link_libraries(libprotobuf-lite PRIVATE ${log-lib})
 endif()
 target_include_directories(libprotobuf-lite PUBLIC ${protobuf_SOURCE_DIR}/src)
 if(protobuf_BUILD_SHARED_LIBS)
diff --git a/cmake/libprotobuf.cmake b/cmake/libprotobuf.cmake
index 07e4bcf57..041f26fe0 100644
--- a/cmake/libprotobuf.cmake
+++ b/cmake/libprotobuf.cmake
@@ -118,7 +118,8 @@ if(protobuf_LINK_LIBATOMIC)
   target_link_libraries(libprotobuf PRIVATE atomic)
 endif()
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
-  target_link_libraries(libprotobuf PRIVATE log)
+  find_library(log-lib log )
+  target_link_libraries(libprotobuf PRIVATE ${log-lib})
 endif()
 target_include_directories(libprotobuf PUBLIC ${protobuf_SOURCE_DIR}/src)
 if(protobuf_BUILD_SHARED_LIBS)
